### DISCUSSION TITLE
How to filter a boolean select with setCondition

### DISCUSSION THREAD

**Discussion by abcpremium** [pimcore/pimcore#12943] [Q&A / Support] [answered]
This question might be a bit stupid, but I struggle for quite some time now. I want to filter data object by a boolean select field. But it always returns 0 results.

Here is my query:

```
$entries = new Listing();
$entries->setCondition("hasOrder = ?", [0]);
$entries->load();
```

I tried the query values `0`, `false`, `'false'`, `'No'`, `'no'`, nothing works...

Can somebody help me how to get all with a false value? Thank you in advance!

**Comment by NiklasBr** [comment, marked as answer]
"Yes" should be `1` in the database, "No" should be `-1` and "Empty" should be `NULL`


```php
$entries->setCondition("hasOrder IS NULL");
$entries->setCondition("hasOrder = 1");
$entries->setCondition("hasOrder = -1");
```

**Reply by dvesh3** [reply]
docs here https://pimcore.com/docs/pimcore/current/Development_Documentation/Objects/Object_Classes/Data_Types/Others.html#page_Boolean-Select
